R/GQDesign.R

```diff
@@ -0,0 +1,81 @@
+setClass("GQDesign", representation(
+        n1        = "numeric",
+        c1f       = "numeric",
+        c1e       = "numeric",
+        n2_pivots = "numeric",
+        c2_pivots = "numeric",
+        knots     = "numeric",
+        weights   = "numeric"
+    ),
+    contains = "Design")
+
+GQDesign <- function(n1, c1f, c1e, n2_pivots, c2_pivots, knots, weights) {
+    if (length(n2_pivots) != length(knots) | length(c2_pivots) != length(knots) |
+        length(weights) != length(knots) )
+        stop("length of pivot vectors does not fit")
+    new("GQDesign", n1 = n1, c1f = c1f, c1e = c1e, n2_pivots = n2_pivots,
+        c2_pivots = c2_pivots, knots = knots, weights = weights)
+}
+
+setMethod("update", signature("GQDesign"),
+    function(object, params, ...) {
+        new("GQDesign",
+            n1 = params[1],
+            c1f = params[2],
+            c1e = params[3],
+            n2_pivots = params[4:(3 + length(object@knots))],
+            c2_pivots = params[(4 + length(object@knots)):(length(params))],
+            knots = object@knots,
+            weights = object@weights)
+    })
+
+setMethod("n1", signature("GQDesign"), function(d, ...) d@n1)
+
+
+setGeneric("get_knots", function(d, ...) standardGeneric("get_knots"))
+setMethod("get_knots", signature("GQDesign"),
+    function(d, ...){
```

Comment:
you can get the knots directly from the rule by `d@rule$x`, just need to scale them appropriately

Reply:
But isn't it nice to have a function that does this automatically?

Reply:
what do u mean by automatically?
For me, it is all about consistency when u use gaussquad why not use their exact knots as well? In what way is that non-automatic?

Reply:
oh I see, you did exactly what i meant. I was't suggesting that people fiddle with d@rule$x automatically, but yoour previous implementation did not use a rule object since being more general...